tests/test_list_associated_files.py

```diff
@@ -0,0 +1,165 @@
+
+    },
+    {  # p4: Only .nfo associated file allowed
+        'path': 'media/postprocess/',
+        'structure': (
+                'Show.S01E01.720p.HDTV.X264-DIMENSION.mkv',
+                'Show.S01E01.720p.HDTV.X264-DIMENSION.pt-BR.srt',
+                'Show.S01E01.720p.HDTV.X264-DIMENSION.nfo',
+                'Show.S01E01.720p.HDTV.X264-DIMENSION.sfv'
+        ),
+        'expected_associated_files': ['Show.S01E01.720p.HDTV.X264-DIMENSION.nfo',
+                                      ],
+        'allowed_extensions': "nfo"
+    },
+    {  # p5: No allowed extensions
+        'path': 'media/postprocess/',
+        'structure': (
+                'Show.S01E01.720p.HDTV.X264-DIMENSION.mkv',
+                'Show.S01E01.720p.HDTV.X264-DIMENSION.pt-BR.srt',
+                'Show.S01E01.720p.HDTV.X264-DIMENSION.nfo',
+                'Show.S01E01.720p.HDTV.X264-DIMENSION.sfv'
+        ),
+        'expected_associated_files': [],
+        'allowed_extensions': ""
+    },
+    {  # p6: Associated file based on RARed file
+        'path': 'media/postprocess/',
+        'structure': (
+                'Show.S01E01.720p.HDTV.X264-DIMENSION.mkv',
+                'show.101.720p-dimension.rar',
+                'show.101.720p-dimension.nfo'
+        ),
+        'expected_associated_files': ['show.101.720p-dimension.nfo'],
+        'allowed_extensions': "nfo"
+    },
+    {  # p7: 'Subtitles only' param for associated files
+        'path': 'media/postprocess/',
+        'structure': (
+                'Show.S01E01.720p.HDTV.X264-DIMENSION.mkv',
+                'Show.S01E01.720p.HDTV.X264-DIMENSION.pt-BR.srt',
+                'Show.S01E01.720p.HDTV.X264-DIMENSION.nfo',
+                'Show.S01E01.720p.HDTV.X264-DIMENSION.sfv'
+        ),
+        'expected_associated_files': ['Show.S01E01.720p.HDTV.X264-DIMENSION.pt-BR.srt'],
+        'allowed_extensions': "nfo,srt",
+        'subtitles_only': True
+    },
+    {  # p8: Subtitle in subfolder. Check subfolders enabled
+        'path': 'media/postprocess/',
+        'structure': (
+                'Show.S01E01.720p.HDTV.X264-DIMENSION.mkv',
+                {'subs': (
+                    'RARBG.txt', 'Show.S01E01.720p.HDTV.X264-DIMENSION.pt-BR.srt',
+                )}
+        ),
+        'expected_associated_files': ['Show.S01E01.720p.HDTV.X264-DIMENSION.pt-BR.srt',
+                                      ],
+        'allowed_extensions': "nfo,srt",
+        'subfolders': True
+
+    },
+])
+def test_list_associated_files(p, create_structure):
+    """Run the test."""
+    # Given
+    test_path = create_structure(p['path'], structure=p['structure'])
+    path = os.path.join(test_path, os.path.normcase(p['path']))
+    media_file = os.path.join(path, p['structure'][0])
+    expected_associated_files = p['expected_associated_files']
+    subtitles_only = p.get('subtitles_only', False)
+    subfolders = p.get('subfolders', False)
+    app.ALLOWED_EXTENSIONS = p['allowed_extensions']
```

Comment:
This should be  monkeypatched to avoid  interference with other tests. Test  should change state and leave it behind. 

`monkeypatch.setattr(app, 'NAME', value)`